src/utils/styled_lines_iterator.rs

```diff
@@ -0,0 +1,224 @@
+use utils::LinesIterator;
+use style::*;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Generates rows of styled text in constrained width.
+///
+/// Given styled text and a width constraint, it iterates over
+/// collections of styled spans, each making up a row within the constraint.
+///
+/// # Examples
+///
+/// ```
+/// # use cursive::style::{StyledString, StyledText};
+/// # use cursive::utils::StyledLinesIterator;
+/// # let text_styled: &[StyledString] = &[].as_ref();
+/// let text_plain = text_styled.to_plain();
+/// let spans = StyledLinesIterator::make_spans(text_styled);
+/// // You can now keep `text_plain` and `spans` alive while using `iter`.
```

Comment:
Regarding the `compute_rows` signature I posted, it is indeed overly simplistic, and was just meant to show an example of input/output flow.

Though thinking about it again, we may need 2 structures after all: the width only needs to be computed when creating the rows, so the input for the updated `LineIterator` could be a list of

```rust
struct StyledStr<'a> {
    content: Cow<'a, str>,
    style: Style
}
```
and the output would use:
```rust
struct StyledSpan<'a> {
    content: Cow<'a, str>,
    style: Style,
    width: usize,
}
```
We'll need the width multiple times and don't want to compute it each time, but we don't need it (and should not compute it) before computing the rows. Having a single type would be a bit ugly, since the `width` field would just be left empty by the parser, and would be ignored by the `LineIterator`, so having two types makes sense.